From f3ed4276582b77feb67364c263b3fced5e70ae8b Mon Sep 17 00:00:00 2001
From: Sanghee Kim <sh0130.kim@samsung.com>
Date: Mon, 17 Jun 2019 13:25:58 +0900
Subject: [PATCH] mbedtls: config: set MBEDTLS_NO_UDBL_DIVISION to improve
 performance

This feature is to speed up some operations on bignum regarding
double-width division.
However it takes a long time on esp8266 then disable this.

Before (9 secs),
I ( 2735) [IoT]: app_main(228) > + SIGN
I (11865) [IoT]: app_main(232) > - SIGN

After (5 secs),
I (2734) [IoT]: app_main(228) > + SIGN
I (7864) [IoT]: app_main(232) > - SIGN

Change-Id: I2eadd9503b213550d4836fc3b5542bf292183abc
Signed-off-by: Sanghee Kim <sh0130.kim@samsung.com>
---
 components/ssl/mbedtls/port/esp8266/include/mbedtls/esp_config.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/components/ssl/mbedtls/port/esp8266/include/mbedtls/esp_config.h b/components/ssl/mbedtls/port/esp8266/include/mbedtls/esp_config.h
index fb6354d..1db5eb9 100644
--- a/components/ssl/mbedtls/port/esp8266/include/mbedtls/esp_config.h
+++ b/components/ssl/mbedtls/port/esp8266/include/mbedtls/esp_config.h
@@ -84,7 +84,7 @@
  * example, if double-width division is implemented in software, disabling
  * it can reduce code size in some embedded targets.
  */
-//#define MBEDTLS_NO_UDBL_DIVISION
+#define MBEDTLS_NO_UDBL_DIVISION
 
 /**
  * \def MBEDTLS_HAVE_SSE2
-- 
2.7.4

